test/Atc.Rest.ApiGenerator.Tests/SyntaxGenerators/Api/ContractInterfaceTest.cs

```diff
@@ -0,0 +1,48 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Atc.Rest.ApiGenerator.Models;
+using Atc.Rest.ApiGenerator.SyntaxGenerators;
+using Atc.Rest.ApiGenerator.SyntaxGenerators.Api;
+using Atc.Rest.ApiGenerator.Tests.Helpers.CodeGenerator;
+using Microsoft.OpenApi.Models;
+using VerifyXunit;
+using Xunit;
+
+namespace Atc.Rest.ApiGenerator.Tests.SyntaxGenerators
+{
+    [UsesVerify]
+    public class ContractInterfaceTest : SyntaxGeneratorTestBase
+    {
+        public static IEnumerable<object[]> YamlFiles { get; } = AllFiles
+            .Where(x => x.FilePath.Contains("ContractInterface", System.StringComparison.Ordinal))
+            .Select(x => new object[] { x });
+
+        protected override ISyntaxCodeGenerator CreateApiGenerator(ApiProjectOptions apiProject)
+        {
+            // Verify spec file suported for unit test
+            Assert.Single(apiProject.Document.Paths);
+            var urlPath = apiProject.Document.Paths.First();
+            Assert.False(urlPath.IsPathStartingSegmentName(FocusOnSecment));
+            Assert.Single(urlPath.Value.Operations);
+            var urlOperation = urlPath.Value.Operations.First();
```

Comment:
```suggestion
            var (operationType, openApiOperation) = urlPath.Value.Operations.First();
```

And implement necessary changes below after deconstruction.